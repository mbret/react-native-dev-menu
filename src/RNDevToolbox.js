/**
 * @flow
 */
import React, { PureComponent } from 'react'
import { style } from './constants'
import { RNDevToolboxDev } from './RNDevToolboxDev'
import { RNDevToolboxProd } from './RNDevToolboxProd'
import type { RNDevToolboxProps } from './RNDevToolboxBase'

export class RNDevToolbox extends PureComponent<RNDevToolboxProps<{}>> {
  static VALID = style.colors.colorValid
  static DANGER = style.colors.colorError

  useProd = () => {
    // forced enabled
    if (this.props.enable === true) return false

    if (this.props.enable === false) return true

    // prod env ?
    return !__DEV__
  }

  render () {
    return this.useProd() ? <RNDevToolboxProd {...this.props} /> : <RNDevToolboxDev {...this.props} />
  }
}
