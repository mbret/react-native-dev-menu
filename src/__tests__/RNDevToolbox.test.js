/* eslint-env jest */
/**
 * @flow
 */
import React from 'react'
import { RNDevToolbox } from '../RNDevToolbox'
import renderer from 'react-test-renderer'

describe('<RNDevToolbox>', () => {
  test('renders correctly', () => {
    const a = renderer.create(<RNDevToolbox />).toJSON()
    expect(a).toMatchSnapshot()

    const b = renderer.create(<RNDevToolbox enable={false} />).toJSON()
    expect(b).toMatchSnapshot()
  })
})
