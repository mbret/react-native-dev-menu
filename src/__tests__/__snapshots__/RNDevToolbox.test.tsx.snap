// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RNDevToolbox> renders correctly 1`] = `
<View
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "padding": 10,
          "paddingBottom": -10,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#dcdcdcd6",
              "margin": 1,
              "padding": 2,
            }
          }
        >
          <Text>
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Open developer menu
              :
            </Text>
             
            Check gesture on real device. Command+D on iOS. Ctrl+M on Android
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#dcdcdcd6",
              "marginTop": 10,
              "opacity": 1,
              "padding": 5,
            }
          }
        >
          <Text>
            Hide Modal
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4a4a4ad6",
          "height": 20,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "right": 5,
          "top": 5,
          "width": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;

exports[`<RNDevToolbox> renders correctly 2`] = `null`;

exports[`<RNDevToolbox> renders correctly 3`] = `
<View
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "padding": 10,
          "paddingBottom": -10,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#dcdcdcd6",
              "margin": 1,
              "padding": 2,
            }
          }
        >
          <Text>
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Open developer menu
              :
            </Text>
             
            Check gesture on real device. Command+D on iOS. Ctrl+M on Android
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#dcdcdcd6",
              "marginTop": 10,
              "opacity": 1,
              "padding": 5,
            }
          }
        >
          <Text>
            Hide Modal
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4a4a4ad6",
          "height": 20,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "right": 5,
          "top": 5,
          "width": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
